Sources/NIOCore/Interfaces.swift

```diff
@@ -62,6 +64,7 @@ private extension ifaddrs {
 }
 #endif
 
+#if !os(WASI)
```

Comment:
Can we keep the type itself available but mark `init` unavailable to reduce subsequent ifdef?